ZFSin/spl/module/spl/spl-vmem.c

```diff
@@ -2139,9 +2139,9 @@ vmem_destroy(vmem_t *vmp)
 
 	kstat_delete(vmp->vm_ksp);
 
+	vmem_free(vmem_vmem_arena, vmp, sizeof (vmem_t));
 	mutex_destroy(&vmp->vm_lock);
 	cv_destroy(&vmp->vm_cv);
-	vmem_free(vmem_vmem_arena, vmp, sizeof (vmem_t));
```

Comment:
Isn't that freeing the memory of `vmp` then dereferencing it in `vmp->vm_lock` ?

Reply:
oh yeah, you could be right. But never got a crash there though 😏. Should take a closer look and change that back maybe

Reply:
hm, so as it seems vmem_xfree gets called from here most times... therefore we don't have to deal with freed memory. But it also works the other way round for me, so maybe just leave that as it is for the moment